src/util/db.ts

```diff
@@ -0,0 +1,30 @@
+export interface CouchConnectionInfo {
+  uri: string
+  hostname: string
+  username: string
+  password: string
+  database: string
+  auth: string
+}
+
+export function dbUrlInfo(url: string): CouchConnectionInfo {
+  const {
+    hostname,
+    username,
+    password,
+    pathname: database,
+    protocol,
+    port
+  } = new URL(url)
+  const uri = `${protocol}//${username}:${password}@${hostname}:${port}`
+  const auth = Buffer.from(`${username}:${password}`).toString('base64')
+
+  return {
+    uri,
+    username,
+    password,
+    hostname,
+    database,
+    auth
+  }
+}
```

Comment:
how is that different than:

``` 
export function dbUrlInfo(uri: string): CouchConnectionInfo {
  const { hostname, username, password, pathname: database } = new URL(uri)

  return {
    uri,
    username,
    password,
    hostname,
    database,
    auth: Buffer.from(`${username}:${password}`).toString('base64')
  }
}
```

Reply:
Semantically, the only difference is that you're including the path in the `uri` output. Everything else is just slightly terser.

Reply:
Perhaps I should change the field from `uri` to `couchUri` to make it clearer what it is (the base URI)